src/main/java/com/fasterxml/jackson/core/JsonFactory.java

```diff
@@ -2199,8 +2187,9 @@ protected IOContext _createContext(ContentReference contentRef, boolean resource
      */
     @Deprecated // @since 2.13
     protected IOContext _createContext(Object rawContentRef, boolean resourceManaged) {
+        BufferRecyclerPool pool = _getBufferRecyclerPool();
         return new IOContext(_streamReadConstraints, _streamWriteConstraints, _errorReportConfiguration,
-                _getBufferRecycler(),
```

Comment:
Alas, I don't like this, due to backwards-(in)compatibility aspect -- this will immediately break usage by those `JsonFactory` sub-classes that override this method (I think CSV one does, probably others).
Or rather, require synchronized changes (technically allowed as per my "internal vs external" compatibility goals.. as long as all 2.16 components work together).